embassy-mcxa/src/trng.rs

```diff
@@ -0,0 +1,620 @@
+#[non_exhaustive]
+pub struct Config {
+    /// Total number of Entropy samples that will be taken during
+    /// Entropy generation.
+    pub sample_size: u16,
+
+    /// Length (in system clocks) of each Entropy sample taken.
+    pub entropy_delay: u16,
+
+    /// Enable or disable internal self-tests.
+    pub self_test: SelfTest,
+
+    /// Frequency Counter Maximum Limit
+    pub freq_counter_max: u32,
+
+    /// Frequency Counter Minimum Limit
+    pub freq_counter_min: u32,
+
+    /// Statistical check monobit max limit
+    pub monobit_limit_max: u16,
+
+    /// Statistical check monobit min limit
+    pub monobit_limit_min: u16,
+
+    /// Statistical check run length 1 limit max
+    pub run_length1_limit_max: u16,
+
+    /// Statistical check run length 1 limit min
+    pub run_length1_limit_min: u16,
+
+    /// Statistical check run length 2 limit max
+    pub run_length2_limit_max: u16,
+
+    /// Statistical check run length 2 limit min
+    pub run_length2_limit_min: u16,
+
+    /// Statistical check run length 3 limit max
+    pub run_length3_limit_max: u16,
+
+    /// Statistical check run length 3 limit min
+    pub run_length3_limit_min: u16,
+
+    /// Statistical check run length 4 limit max
+    pub run_length4_limit_max: u16,
+
+    /// Statistical check run length 4 limit min
+    pub run_length4_limit_min: u16,
+
+    /// Statistical check run length 5 limit max
+    pub run_length5_limit_max: u16,
+
+    /// Statistical check run length 5 limit min
+    pub run_length5_limit_min: u16,
+
+    /// Statistical check run length 6 limit max
+    pub run_length6_limit_max: u16,
+
+    /// Statistical check run length 6 limit min
+    pub run_length6_limit_min: u16,
+
+    /// Retry count
+    pub retry_count: u8,
+
+    /// Long run limit max
+    pub long_run_limit_max: u8,
+
+    /// Sparse bit limit
+    pub sparse_bit_limit: u16,
+
+    /// Poker limit max
+    pub poker_limit_max: u32,
+
+    /// Oscillator mode
+    pub osc_mode: OscMode,
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            sample_size: 1024,
+            entropy_delay: 32_000,
+            self_test: SelfTest::Enabled,
+            freq_counter_max: 75_000,
+            freq_counter_min: 30_000,
+            monobit_limit_max: 596,
+            monobit_limit_min: 427,
+            run_length1_limit_max: 187,
+            run_length1_limit_min: 57,
+            run_length2_limit_max: 105,
+            run_length2_limit_min: 28,
+            run_length3_limit_max: 97,
+            run_length3_limit_min: 33,
+            run_length4_limit_max: 0,
+            run_length4_limit_min: 0,
+            run_length5_limit_max: 0,
+            run_length5_limit_min: 0,
+            run_length6_limit_max: 0,
+            run_length6_limit_min: 0,
+            retry_count: 2,
+            long_run_limit_max: 32,
+            sparse_bit_limit: 0,
+            poker_limit_max: 0,
+            osc_mode: OscMode::DualOscs,
```

Comment:
Should I add a comment here stating these were directly taken from NXP's sdk sample?

Reply:
Could do that, or make this a method like `Config::nxp_sdk_defaults()`, document it there, and call that func inside of `Default::default`.

Reply:
I added three dedicated constructors following the manual and left a `new_with_custom_config` constructor which takes the full `Config` argument and allows the user to configure it any way they want. Let me know what you think? Happy to update.